src/main/java/com/devcycle/sdk/server/local/api/DVCLocalClient.java

```diff
@@ -113,40 +110,43 @@ public <T> Variable<T> variable(User user, String key, T defaultValue) {
             .isDefaulted(true)
             .build();
 
-    if (!isInitialized) {
+    if (!configManager.isConfigInitialized() || !isInitialized) {
+      System.out.println("Variable called before DVCClient has initialized, returning default value");
+      try {
+        eventQueueManager.queueAggregateEvent(Event.builder().type("aggVariableDefaulted").target(key).build(), null);
+      } catch (Exception e) {
+        System.out.printf("Unable to parse aggVariableDefaulted event for Variable %s due to error: %s", key, e.toString());
+      }
       return defaultVariable;
     }
-
+    String variableJSON = null;
     try {
-      BucketedUserConfig bucketedUserConfig = localBucketing.generateBucketedConfig(sdkKey, user);
-      if (bucketedUserConfig.variables.containsKey(key)) {
-        BaseVariable baseVariable = bucketedUserConfig.variables.get(key);
+      variableJSON = localBucketing.getVariable(sdkKey, user, key, variableType, true);
+      if (variableJSON == null || variableJSON.isEmpty()) {
+        return defaultVariable;
+      } else {
+        ObjectMapper mapper = new ObjectMapper();
+        Variable baseVariable = mapper.readValue(variableJSON, Variable.class);
+
         Variable<T> variable = (Variable<T>) Variable.builder()
-          .key(key)
-          .type(baseVariable.getType())
-          .value(baseVariable.getValue())
-          .defaultValue(defaultValue)
-          .isDefaulted(false)
-          .build();
+                .key(key)
+                .type(baseVariable.getType())
+                .value(baseVariable.getValue())
+                .defaultValue(defaultValue)
+                .isDefaulted(false)
+                .build();
         if (variable.getType() != variableType) {
-          throw new IllegalArgumentException("Variable type mismatch, returning default value");
+          System.out.printf("Variable type mismatch, returning default value");
```

Comment:
We should add supporting over-writeable loggers, like we have in other SDKs, to the list of features we need here. Is there a way to tag this with a log level so folks who don't want it to print can set the log level?

Reply:
Agreed. I have an open ticket to review how error handling is done in this SDK.  Everything is System.out which is problematic